roles/network/tasks/restart.yml

```
@@ -81,10 +81,13 @@
     state: restarted
   when: hostapd_enabled and wifi_slave.stdout is defined and wifi_slave.stdout == 0
 
+#both interfaces.d and systemd-networkd should have br0 available and Appliance lacks br0
 - name: User choice of dnsmasq or dhcpd - restarting {{ dhcp_service2 }}
   systemd:
     name: "{{ dhcp_service2 }}"
     state: restarted
-  when: not no_net_restart and discovered_wireless_iface == "none"
+  when: (not no_net_restart and not wifi_up_down) or wifi_up_down
```

Redundant Logic...can be more concise:
```suggestion
  when: (not no_net_restart) or wifi_up_down
```

---

If you think so, should this causes any issues with dnsmanq I will step back and let you discover the reason why if failed and not answer any issue raised about the failure.